lib/src/command/deps.dart

```diff
@@ -231,4 +263,61 @@ class DepsCommand extends PubCommand {
         'was generated, please run "pub get" again.');
     return null;
   }
+
+  /// Outputs all executables reachable from [entrypoint].
+  void _outputExecutables() {
+    var packages = []
+      ..add(entrypoint.root)
+      ..addAll((_includeDev
+              ? entrypoint.root.immediateDependencies
+              : _nonDevDependencies(entrypoint.root))
+          .map((dep) => entrypoint.packageGraph.packages[dep.name]));
+
+    for (var package in packages) {
+      var executables = _getExecutablesFor(package);
+      if (executables.isNotEmpty) {
+        _buffer.writeln(_formatExecutables(package.name, executables.toList()));
+      }
+    }
+  }
+
+  /// Lists all Dart files in the `bin` directory of the [package].
+  ///
+  /// Returns file names without extensions.
+  List<String> _getExecutablesFor(Package package) => package
+      .listFiles(beneath: 'bin', recursive: false)
```

Comment:
There is already such a method in `Package` - `executableIds`. While I'm going to adapt `_getExecutablesFor` to use it, should `_isDartExecutable` be a part of executableIds`?